packages/babel-plugin-proposal-decorators/test/fixtures/2023-11-misc/class-not-available-during-decorators/exec.js

```diff
@@ -6,7 +6,7 @@ function dec(fn) {
 }
 
 @dec(() => {
-  expect(() => Foo).toThrow(ReferenceError);
+  expect(() => Foo.x).toThrow();
```

Comment:
This changes our behavior from a TDZ error to `Foo` being `undefined`, because it now runs inside the static block. In practice we are never very good with preserving TDZ errors, so it should be fine.